Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -287,6 +341,127 @@ private extension ProductStore {
+        for readOnlyProductVariation in readOnlyProductVariations {
+            let storageProductVariation = storage.loadProductVariation(siteID: readOnlyProductVariation.siteID,
+                                                                       productID: readOnlyProductVariation.productID,
+                                                                       variationID: readOnlyProductVariation.variationID) ??
+                storage.insertNewObject(ofType: Storage.ProductVariation.self)
+
+            storageProductVariation.update(with: readOnlyProductVariation)
+            handleProductVariationDimensions(readOnlyProductVariation, storageProductVariation, storage)
+            handleProductVariationImage(readOnlyProductVariation, storageProductVariation, storage)
+            handleProductVariationAttributes(readOnlyProductVariation, storageProductVariation, storage)
+        }
+    }
+
+    /// Updates or inserts the provided StorageProductVariation's dimensions using the provided read-only ProductVariation's dimensions
+    ///
+    func handleProductVariationDimensions(_ readOnlyProductVariation: Networking.ProductVariation,
+                                          _ storageProductVariation: Storage.ProductVariation,
+                                          _ storage: StorageType) {
+
+        if let existingStorageDimensions = storageProductVariation.dimensions {
+            existingStorageDimensions.update(with: readOnlyProductVariation.dimensions)
+        } else {
+            let newStorageDimensions = storage.insertNewObject(ofType: Storage.ProductVariationDimensions.self)
+            newStorageDimensions.update(with: readOnlyProductVariation.dimensions)
+            storageProductVariation.dimensions = newStorageDimensions
+        }
+    }
+
+    /// Updates or inserts the provided StorageProductVariation's image using the provided read-only ProductVariation's image
+    ///
+    func handleProductVariationImage(_ readOnlyProductVariation: Networking.ProductVariation,
+                                     _ storageProductVariation: Storage.ProductVariation,
+                                     _ storage: StorageType) {
+
+        if let existingStorageImage = storageProductVariation.image {
+            if let readOnlyImage = readOnlyProductVariation.image {
+                // The existing storageImage and readOnlyImage both exist so update!
+                existingStorageImage.update(with: readOnlyImage)
+            } else {
+                // readOnlyImage is nil, so nuke the existing storageImage!
+                storageProductVariation.image = nil
+                storage.deleteObject(existingStorageImage)
+            }
+        } else if let readOnlyImage = readOnlyProductVariation.image {
+            // There's a readOnlyImage, but no existing storageImage, so create a new one and save it
```

Comment:
Thanks for the comments, otherwise it is a bit difficult (at least for me) to follow the logic